deploy/AllocatorInit.sol

```diff
@@ -0,0 +1,278 @@
+}
+
+interface VaultLike {
+    function ilk() external view returns (bytes32);
+    function roles() external view returns (address);
+    function buffer() external view returns (address);
+    function vat() external view returns (address);
+    function nst() external view returns (address);
+    function file(bytes32, address) external;
+    function draw(uint256) external;
+    function wipe(uint256) external;
+}
+
+interface BufferLike {
+    function approve(address, address, uint256) external;
+}
+
+interface SwapperLike {
+    function roles() external view returns (address);
+    function ilk() external view returns (bytes32);
+    function buffer() external view returns (address);
+    function swap(address, address, uint256, uint256, address, bytes calldata) external;
+}
+
+interface DepositorUniV3Like {
+    struct LiquidityParams {
+        address gem0;
+        address gem1;
+        uint24  fee;
+        int24   tickLower;
+        int24   tickUpper;
+        uint128 liquidity;
+        uint256 amt0Desired;
+        uint256 amt1Desired;
+        uint256 amt0Min;
+        uint256 amt1Min;
+    }
+
+    struct CollectParams {
+        address gem0;
+        address gem1;
+        uint24  fee;
+        int24   tickLower;
+        int24   tickUpper;
+    }
+
+    function roles() external view returns (address);
+    function ilk() external view returns (bytes32);
+    function buffer() external view returns (address);
+    function deposit(LiquidityParams memory) external returns (uint128, uint256, uint256);
+    function withdraw(LiquidityParams memory, bool) external returns (uint256, uint256);
+    function collect(CollectParams memory) external returns (uint256, uint256);
+}
+
+interface StableSwapperLike {
+    function swapper() external view returns (address);
+}
+
+interface StableDepositorUniV3Like {
+    function depositor() external view returns (address);
+}
+
+interface ConduitMoverLike {
+    function ilk() external view returns (bytes32);
+    function buffer() external view returns (address);
+}
+
+interface KissLike {
+    function kiss(address) external;
+}
+
+struct AllocatorConfig {
+    uint256 debtCeiling;
+    address allocatorProxy;
+    uint8 facilitatorRole;
+    uint8 automationRole;
+    address facilitator;
+    address stableSwapperKeeper;
+    address stableDepositorUniV3Keeper;
+    address conduitMoverKeeper;
+    address[] swapTokens;
+    address[] depositTokens;
+    bytes32 vaultClKey;
+    bytes32 bufferClKey;
+    address ilkRegistry;
+    string ilkRegistryName;
+    string ilkRegistrySymbol;
+}
+
+library AllocatorInit {
+    uint256 constant WAD = 10 ** 18;
+    uint256 constant RAY = 10 ** 27;
+    uint256 constant RAD = 10 ** 45;
+
+    function switchOwner(address base, address currentOwner, address newOwner) internal {
+        if (currentOwner == newOwner) return;
+        require(WardsLike(base).wards(currentOwner) == 1, "AllocatorInit/current-owner-not-authed");
```

Comment:
Do we really need these two?

Reply:
Yeah that `switchOwner` was copied from the ScriptTools's `switchOwner`.

In theory we can do what you suggest, but note that it should be "deny(address(this))".
Potentially we can change also `RolesLike(sharedInstance.roles).setIlkAdmin(ilk, networkInstance.owner);` to
`RolesLike(sharedInstance.roles).setIlkAdmin(ilk, address(this));`
And if we do all of that we can remove `owner` from `AllocatorNetworkInstance`.

If we do that we'll need to change the test to do a delegateCall (if we use `address(this)` or `msg.sender` and want to check them correctly). Currently we just prank to the pause proxy, which makes sure external calls from the emulated spell are done with the right `msg.sender`, but it does not emulate `address(this)` or `msg.sender` in the emulated spell correctly.

Reply:
Let's leave as it is to avoid further complications.

Reply:
Unresolving this one just to keep a reminder in case we get some response from Foundry team. If nothing changes in the short term I think we should leave it as it is.

Reply:
Done.